Compile with Java 7

Saw this exception in an internal project:

```
Caused by: java.lang.UnsupportedClassVersionError: com/palantir/launchconfig/LaunchConfigPlugin : Unsupported major.minor version 52.0
    at org.gradle.api.internal.plugins.DefaultPluginRegistry$1.load(DefaultPluginRegistry.java:71)
    at org.gradle.api.internal.plugins.DefaultPluginRegistry$1.load(DefaultPluginRegistry.java:51)
    at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3524)
    at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2317)
    at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2280)
```

Do you plan to support Java 7 with this plugin or do users need to run their Gradle builds with Java 8?

--- a/circle.yml
+++ b/circle.yml
@@ -1,6 +1,6 @@
 machine:
   java:
-    version: oraclejdk8
+    version: oraclejdk7
   environment:
     TERM: dumb
 

